src/python/pants/backend/native/tasks/native_compile.py

```diff
@@ -181,9 +180,23 @@ def _make_compile_request(self, versioned_target, dependencies, external_libs_pr
       compiler_options=(self._compile_settings
                             .native_build_step_settings
                             .get_merged_args_for_compiler_option_sets(compiler_option_sets)),
-      output_dir=versioned_target.results_dir)
-
-  def _make_compile_argv(self, compile_request):
+      output_dir=versioned_target.results_dir,
+      header_file_extensions=self._compile_settings
+                                 .header_file_extensions)
+
+  def _split_sources_and_headers(self, sources, header_file_extensions):
+    sources_for_compiler = []
+    header_files = []
+    for s in sources:
+      for ext in header_file_extensions:
+        if s.endswith(ext):
```

Comment:
Little-known, though documented, api: `endswith` can take a tuple.

Reply:
Changed, and noted!